### ISSUE TITLE
Catch CssSyntaxError and report it alongside stylelint warnings

### ISSUE THREAD

**Issue by PhilippBaschke** [stylelint/stylelint#883]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I am trying to integrate the stylelint checker with flycheck (see: https://github.com/flycheck/flycheck/pull/903). Since the code is checked while typing, **syntax errors** in the code that is passed to stylelint are **not uncommon**. As far as I can tell, `CssSyntaxErrors` are thrown, which also results in an Error in flycheck instead of highlighting the offending line and column in the editor. 

> What CSS is needed to reproduce this issue?

```css
/* style.css */
h1 {
  color: 'red; /* syntax error: Unclosed quote */
    margin: 0 auto;
}
```

> Which version of stylelint are you using?

`4.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint < style.css`

> What did you expect to happen?

A normal rule violation, probably with error severity `error`:

```shell
<input css 1>
2:10	Unclosed quote (syntax error)
```

> What actually happened (e.g. what warnings or errors you are getting)?

A `CssSyntaxError` is thrown:

```shell
CssSyntaxError: <css input>:2:10: Unclosed quote
    ...Stacktrace (omitted)
```

**Comment by davidtheclark** [comment]
Could this be something done on the side of your plugin (have it deal the same way with CssSyntaxErrors as it does with stylelint warnings), instead of changing the behavior of stylelint?

**Comment by PhilippBaschke** [comment]
@lunarjorn Sorry for dragging you into this, but you might be able to answer this question better than I ever could. Thank you!

**Comment by lunaryorn** [comment]
@PhilippBaschke Speaking as the maintainer of said plugin, I'd prefer to have a consistent output format.  The main issue is that even if we explicitly select JSON as output format, we still get syntax errors as plain text which is somewhat ugly to parse then.

**Comment by PhilippBaschke** [comment]
Thank you for clarifying! :relaxed:

**Comment by davidtheclark** [comment]
Ok. Marking as help wanted for someone to experiment with catching the CssSyntaxError and reporting it alongside stylelint warnings.

**Comment by PhilippBaschke** [comment]
Thank you @davidtheclark. I have a PR in the making.

**Comment by lunaryorn** [comment]
@PhilippBaschke Is there any progress on this issue?

**Comment by PhilippBaschke** [comment]
@lunaryorn Sorry :disappointed: I was really busy but I got some time this week.
